File: code/mathematical_algorithms/src/tower_of_hanoi/tower_of_hanoi.c
#include <stdio.h>
// Part of Cosmos by OpenGenus Foundation
int moves = 0;
void hanoi(int disks, char A, char B, char C);
 
int 
main()
{
    int n; // Number of disks
    printf("Enter the number of disks- ");
    scanf("%d", &n);

    hanoi(n, 'A', 'C', 'B');  // A, B and C are names of rods
    printf("Total moves- %d\n", moves);

    return (0);
}

void 
hanoi(int disks, char A, char B, char C)
{
    moves++;
    if (disks==1) {
        printf("Move disk 1 from rod %c to rod %c \n", A, B);
        return;
    }
    hanoi(disks -1 , A, C, B);
    printf("Move disk %d from rod %c to rod %c \n", disks, A, B);
    hanoi(disks - 1, C, B, A);
}

File: code/mathematical_algorithms/src/tower_of_hanoi/tower_of_hanoi.py
# Part of Cosmos by OpenGenus Foundation
moves = 0


def TowerOfHanoi(disks, A, B, C):
    global moves
    moves += 1
    if disks == 1:
        print("Move disk 1 from rod", A, "to rod", B)
        return
    TowerOfHanoi(disks - 1, A, C, B)
    print("Move disk", disks, "from rod", A, "to rod", B)
    TowerOfHanoi(disks - 1, C, B, A)


# Driver code
n = int(input("Enter the number of disks- "))
TowerOfHanoi(n, "A", "C", "B")
print("Total moves required = " + str(moves))
